python_coreml_stable_diffusion/torch2coreml.py

```diff
@@ -453,6 +471,163 @@ def forward(self, z):
     gc.collect()
 
 
+def convert_vae_encoder(pipe, args):
+    """ Converts the VAE Encoder component of Stable Diffusion
+    """
+    out_path = _get_out_path(args, "vae_encoder")
+    if os.path.exists(out_path):
+        logger.info(
+            f"`vae_encoder` already exists at {out_path}, skipping conversion."
+        )
+        return
+
+    if not hasattr(pipe, "unet"):
+        raise RuntimeError(
+            "convert_unet() deletes pipe.unet to save RAM. "
+            "Please use convert_vae_encoder() before convert_unet()")
+
+    sample_shape = (
+        1,  # B
+        3,  # C (RGB range from -1 to 1)
+        args.latent_h or pipe.unet.config.sample_size * 8,  # H
```

Comment:
This is the shape logic for the `vae_decoder`'s input which would match `vae_encoder` output. If `--latent-h` or `--latent-w` were to be specified, the model will have incorrect input shape for `sample`. The multiplier 8 needs to apply to the output of the `or` operation.